pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -295,6 +311,7 @@ func (c *Controller) deleteShoot(logger *logrus.Entry, shoot *gardencorev1beta1.
 		ctx = context.TODO()
 		err error
 
+		operationType              = gardencorev1beta1helper.ComputeOperationType(shoot.ObjectMeta, shoot.Status.LastOperation)
```

Comment:
Why do you have to compute this here and then pass it in each call to the status update?

Reply:
Because I couldn't use it for the migration scenario, either this or we should add the shoot.spec.seedName and shoot.status.seedName to the ComputeLastOperation